### ISSUE TITLE
Any advice to make it work on RPI?

### ISSUE THREAD

**Issue by SebastienSyd** [nzbart/ArlecRC210Switch#3]
Morning mate. I have the same Arlec RC210 model (bought from Bunnings in AU) and I have Raspberry Pi connected to 433MHz Receiver/Transmitter.
I've been trying to use the same binary codes as yours and it doesn't work (I used rc-switch with the modifications you suggested here: https://github.com/sui77/rc-switch/wiki/Description-of-socket-protocols-from-different-brands-and-models). 
Do you think the binary codes are different between models? 
What tool/module/library did you use to sniff the codes? The libraries that I used just detect noise and don't actually detect the Arlec signal when I press the buttons.

Thanks
Seb

**Comment by SebastienSyd** [comment]
Update: there was a defect with the cheapo receiver that I bought online. I could fix it and could successfully get the codes.
I can confirm the codes are different between remotes.

However I get codes of 35 bytes for all the codes (A, B, C, D and all). Yours were 34 bytes long. Do I need to change anything from rc-switch code in that case?
Finally, how did you measure the pulselength of 385?

Cheers

**Comment by nzbart** [comment]
@SebastienSyd I believe that I used a logic analyser to capture the bitstream, then painstakingly converted this into binary and discovered a pattern from that.

I still use this code with the same switches, and it's working well, but for new work I'm using TP-Link HS110 WiFi smart plugs, which are also reasonably priced. I found the Arlec switches had too many problems, especially if you want to use more than one within a small area. While it seemed good value because you got four plugs for a good price, they were not all usable. The TP-Link smart plugs have a JSON API and can easily be configured and used independently of the TP-Link control software.

I use a Raspberry PI to control the TP-Link switches via the `python-kasa` package.
